docs/shared/packages/powerpack-owners/powerpack-owners-plugin.md

```diff
@@ -1,6 +1,326 @@
 ---
 
-Powerpack owners
+The `@nx/powerpack-owners` plugin extends the CODEOWNERS functionality to allow you to define code ownership based on projects in addition to the standard file-based definitions. It leverages the [`nx sync`](/concepts/sync-generators) command to compile `owners` configuration settings from `nx.json` and project configuration files into valid CODEOWNERS files for [GitHub](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners), [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/set-up-and-use-code-owners/) or [GitLab](https://docs.gitlab.com/ee/user/project/codeowners/).
+
+With this plugin, you can specify code ownership using the same project glob syntax as [`nx run-many`](/nx-api/nx/documents/run-many#examples). This allows you to easily define rules for multiple projects that may not be located in the same directory. Also, the CODEOWNERS rules will not need to be revisited if a project location is changed or a new project is added.
+
+{% callout title="This plugin requires an active Nx Powerpack license" %}
+In order to use `@nx/powerpack-owners`, you need to have an active Powerpack license. If you don't have a license or it has expired, the syncing process will stop working and you'll need to manually maintain your CODEOWNERS file.
+{% /callout %}
+
+## Setup
+
+1. [Activate Powerpack](/recipes/installation/activate-powerpack) if you haven't already
+2. Install the package
+
+   ```shell
+   nx add @nx/powerpack-owners
+   ```
+
+3. Configure Ownership
+
+   Configure the `@nx/powerpack-owners` plugin in the `nx.json` file or in individual project configuration files. Consult the [Owners Configuration Reference](#owners-configuration-reference) section for more details.
+
+4. Configure the [Sync Generator](/concepts/sync-generators) and CI
+
+Make sure that the `@nx/powerpack-owners:sync-codeowners-file` generators is registered as a `globalGenerator` in `nx.json`:
+
+```jsonc {% fileName="nx.json" %}
+{
+  "sync": {
+    "globalGenerators": ["@nx/powerpack-owners:sync-codeowners-file"]
+  }
+}
+```
+
+Add `nx sync:check` to the beginning of the CI process.
+
+```yaml
+- name: Ensure the workspace configuration is in sync
+  run: npx nx sync:check
+```
+
+It is also often helpful to add `nx sync` as a git push hook or git commit hook.
+
+## Owners Configuration Reference
+
+{% tabs %}
+{% tab label="GitHub" %}
+
+```jsonc {% fileName="nx.json" %}
+{
+  // Can be set to true instead of an object to accept all defaults
+  "owners": {
+    // Options are `github`, `bitbucket` or `gitlab`. (Optional) Defaults to `github`
+    "format": "github",
+    // (Optional) Default changes based on format: `.github/CODEOWNERS`, `.bitbucket/CODEOWNERS`, `.gitlab/CODEOWNERS`
+    "outputPath": "CODEOWNERS",
+    // (Optional)
```

Comment:
This shouldn't be optional for github, only for gitlab where sections is supported. Can you verify this, it's a bug if so

Reply:
I thought this was optional because you could define all your ownership rules in project configuration

Reply:
That is a good point!